gym/wrappers/record_episode_statistics.py

```diff
@@ -4,6 +4,71 @@
 import numpy as np
 
 import gym
+from gym.error import NoMatchingInfoStrategy
+from gym.vector.utils import InfoStrategiesEnum
+
+
+class ClassicStatsInfoStrategy:
+    def __init__(self, num_envs: int):
+        self.info = {}
+
+    def add_info(self, infos: dict, env_num: int):
+        self.info = {**self.info, **infos}
+
+    def add_episode_statistics(self, infos: dict, env_num: int):
+        self.info = {**self.info, **infos}
+
+    def get_info(self):
+        return self.info
+
+
+class ClassicVecEnvStatsInfoStrategy:
+    def __init__(self, num_envs: int):
+        self.info = [{} for _ in range(num_envs)]
+
+    def add_info(self, infos: list, env_num: int):
+        self.info[env_num] = {**self.info[env_num], **infos[env_num]}
+
+    def add_episode_statistics(self, infos: dict, env_num: int):
+        self.info[env_num] = {**self.info[env_num], **infos}
+
+    def get_info(self):
+        return tuple(self.info)
+
+
+class BraxVecEnvStatsInfoStrategy:
+    def __init__(self, num_envs: int):
+        self.num_envs = num_envs
+        self.info = {}
+
+    def add_info(self, info: dict, env_num: int):
+        self.info = {**self.info, **info}
+
+    def add_episode_statistics(self, info: dict, env_num: int):
+        episode_info = info["episode"]
+        self.info["episode"] = self.info.get("episode", {})
+        for k in episode_info.keys():
+            info_array = self.info["episode"].get(
+                k, [None for _ in range(self.num_envs)]
+            )
+            info_array[env_num] = episode_info[k]
+            self.info["episode"][k] = info_array
+
+    def get_info(self):
+        return self.info
+
+
+def get_statistic_info_strategy(wrapped_env_strategy: str):
+    strategies = {
```

Comment:
A comment like above on the info strategies, is there a way of adding statistic info strategies

Reply:
refactored as the InfoStrategyFactory :+1: